src/gas.rs

```diff
@@ -153,14 +161,18 @@ pub fn inject_gas_counter(module: elements::Module, rules: &rules::Set) -> eleme
 	let gas_func = module.import_count(elements::ImportCountType::Function) as u32 - 1;
 	let total_func = module.functions_space() as u32;
 	let mut need_grow_counter = false;
+	let mut error = false;
 
 	// Updating calling addresses (all calls to function index >= `gas_func` should be incremented)
 	for section in module.sections_mut() {
 		match section {
 			&mut elements::Section::Code(ref mut code_section) => {
 				for ref mut func_body in code_section.bodies_mut() {
 					update_call_index(func_body.code_mut(), gas_func);
-					inject_counter(func_body.code_mut(), rules, gas_func);
+					if let Err(_) = inject_counter(func_body.code_mut(), rules, gas_func) {
+						error = true;
```

Comment:
Why do we need to collect `error` here? Why don't just return?

Reply:
Cannot move out of borrowed module, it is returned by value

Reply:
Ah, I see